providers/record.rb

```diff
@@ -51,5 +69,20 @@ def delete_record(name, region, zone_id, record_type)
     end
   end
 
-  delete_record(new_resource.name, new_resource.region, new_resource.zone_id, new_resource.type)
+  count = 0
+  while count <= record_wait_tries do
+    begin
```

Comment:
We could possibly refactor these two blocks of code into 1 since the logic is virtually identical; at the same time this is a small enough change that I don't really care.